pyproject.toml

```diff
@@ -1,7 +1,97 @@
 [build-system]
-requires = ["jupyter_packaging~=0.7.9", "jupyterlab>=3,<=4", "setuptools>=40.8.0", "wheel"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling>=1.5.0", "hatch-nodejs-version"]
+build-backend = "hatchling.build"
+
+[project]
+name = "jupytext"
+description = "Jupyter notebooks as Markdown documents, Julia, Python or R scripts"
+license = { file = "LICENSE" }
+authors = [
+    { name = "Marc Wouts", email = "marc.wouts@gmail.com" },
+]
+readme = "README.md"
+requires-python = ">=3.8"
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "License :: OSI Approved :: MIT License",
+    "Environment :: Console",
+    "Framework :: Jupyter",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Topic :: Text Processing :: Markup",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+dependencies = [
+    "nbformat",
+    "pyyaml",
+    "toml",
+    "markdown-it-py>=1.0.0",
+    "mdit_py_plugins",
+]
+dynamic = ["version"]
+
+[project.urls]
+homepage = "https://github.com/mwouts/jupytext"
+source = "https://github.com/mwouts/jupytext"
+documentation = "https://jupytext.readthedocs.io/en/latest/index.html"
+
+[project.optional-dependencies]
+extension = [
+    "jupytext_extension",
```

Comment:
Does the version synchronization matter with the extension? If so is there a way to use the version of the current project here? Otherwise, can you add a check in the `jupyterlab_jupytext` to check the version consistency dynamically?

Reply:
It is a typo, that should be `jupyterlab_jupytext`. In any case, there is _no_ python package `jupyterlab_jupytext` at the moment. We will have first one after this PR gets merged and we make a new release. If we need, we will have to manually pin the version in `optional-dependencies`. I dont think there is a way to get it programatically.

We do not need to synchronize the version of core `jupytext` with the extension. As stated above, once we get `jupyterlab==5`, maybe we will have to pin lower bound of the extension package.

Reply:
Do you think we could make `jupyterlab_jupytext` a main dependency of `jupytext`? Currently when we install `jupytext` we do get the extension too, if possible (e.g. if it does not add too many dependencies) I would like to preserve this.